Code/HighLevelAPI/src/main/java/ie/tcd/wayfinder/highlevel/navigation/controller/NavigationController.java

```diff
@@ -55,13 +62,25 @@ public String index()
 
         if (startLong >180 || startLong < -180 ) return new ResponseEntity<String>("Longitude of Starting point is invalid!",
                       HttpStatus.BAD_REQUEST);
-       
-        HttpResponse response = Library.GET("http://localhost:3000/path", Optional.empty());
         
-        HttpEntity responseEntity = response.getEntity();
-        String responseXml = EntityUtils.toString(responseEntity);
-     
-        return new ResponseEntity<String>(responseXml, HttpStatus.valueOf(response.getStatusLine().getStatusCode()));
+        UserRouteRequest request = new UserRouteRequest(startLat + "," + startLong, destLat + "," + destLong, mode);
+        
+        String jsonRequestContent = objectMapper().writeValueAsString(request);
+                
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put("Content-Type", "application/json");
+       
+        HttpResponse response = Library.POST("http://localhost:8082/api/route", Optional.of(headers), Optional.of(jsonRequestContent));
```

Comment:
Endpoint will be changed + stored in application.properties in new branch once UI, HL-API, LL-Routing API are deployed